net/route/net_fileroute.c

```diff
@@ -636,36 +606,10 @@ int net_lockroute_ipv4(void)
 #ifdef CONFIG_ROUTE_IPv6_FILEROUTE
 int net_lockroute_ipv6(void)
 {
-  pid_t me = getpid();
-  int ret;
-
-  /* Are we already the holder of the lock? */
-
-  if (g_ipv6_holder == me)
-    {
-      /* Yes.. just increment the count of locks held */
-
-      g_ipv6_count++;
-      ret = OK;
-    }
-  else
+  int ret = nxrmutex_lock(&g_ipv6_lock);
+  if (ret < 0)
     {
-      /* No.. wait to get the lock */
-
-      ret = nxsem_wait(&g_ipv6_exclsem);
```

Comment:
Here just a note. Previously it was possible to return from `net_lockroute_ipv6` with `EINTR`, but now we wait uninterruptable

Reply:
maybe we can add a new api like `nxmutex_lock_interruptible` and `nxrmutex_lock_interruptible`？

Reply:
@xiaoxiang781216 what do you think?

Reply:
Except the code error, it isn't expect to interrupt the lock. I think the interrupt is used for event not lock In most case.